src/HyperscanEngine.cpp

```diff
@@ -55,9 +60,35 @@ void HyperscanEngine::compile(const std::vector<Rule> &rules) {
     throw std::bad_alloc();
 }
 
+void HyperscanEngineStream::compile(const std::vector<Rule> &rules) {
+  HyperscanEngine::compile(rules);
+  streams.reset(new hs_stream_t*[nsessions]);
```

Comment:
Use `std::make_unique` and move-assignment to `streams`, instead of using `new` explicitly. See [C++ Core Guidelines R.11](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#a-namerr-newdeletear11-avoid-calling-new-and-delete-explicitly) for justification.